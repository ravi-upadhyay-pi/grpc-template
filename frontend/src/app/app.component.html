{{message}}
<router-outlet></router-outlet>
