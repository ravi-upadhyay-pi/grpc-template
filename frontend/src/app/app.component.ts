import { Component } from '@angular/core';
import { HelloRequest } from '../generated/helloworld_pb';
import { GreeterClient } from '../generated/HelloworldServiceClientPb';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  title = 'frontend';
  greeterClient: GreeterClient;
  message = '';

  constructor() {
    this.greeterClient = new GreeterClient("");
    const request = new HelloRequest();
    request.setName("Ravi");
    this.greeterClient.sayHello(request, null).then(reply => {
      this.message = reply.getMessage();
    });
  }
}
