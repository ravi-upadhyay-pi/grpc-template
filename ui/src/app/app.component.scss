@use "@material/data-table/data-table";

.create-string-container {
    width: 500px;
    margin-top: 24px;
    margin-inline: auto;

    mat-form-field {
        display: block;
    }
}

.existing-strings {
    width: 500px;
    margin-top: 24px;
    margin-inline: auto;

    table {
        width: 100%;
    }
     
    tr.mat-row {
        height: 20px !important;
    }
}
