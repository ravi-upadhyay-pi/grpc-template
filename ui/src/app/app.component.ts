import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
import { String as ProtoString, Void, Page } from '../generated/grpc_template_pb';
import { GrpcTemplateClient } from '../generated/Grpc_templateServiceClientPb';
import { FormControl } from '@angular/forms';
import { BehaviorSubject, Subject } from 'rxjs';
import { PageEvent } from '@angular/material/paginator';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AppComponent implements OnInit {
  readonly client: GrpcTemplateClient;
  readonly newStringId = new FormControl('');
  readonly newStringValue = new FormControl('');
  readonly strings = new BehaviorSubject<Array<ProtoString>>([]);
  readonly existingStringsCount = new BehaviorSubject<number>(0);
  readonly fetchingStrings = new BehaviorSubject(true);

  constructor() {
    this.client = new GrpcTemplateClient("");
  }

  async ngOnInit() {
    const count = (await this.client.getStringsCount(new Void(), null)).getStringsCount();
    this.existingStringsCount.next(count);
    const page = new Page();
    page.setPageNumber(1);
    page.setPageSize(5);
    this.strings.next((await this.client.getStrings(page, null)).getStringList());
    this.fetchingStrings.next(false);
  }

  async saveString() {
    await this.client.saveString(
      new ProtoString()
        .setId(this.newStringId.value ?? '')
        .setString(this.newStringValue.value ?? ''), 
      null);
  }

  async updatePage(event: PageEvent) {
    const request = new Page();
    request.setPageNumber(event.pageIndex);
    request.setPageSize(event.pageSize);
    this.fetchingStrings.next(true);
    this.strings.next((await this.client.getStrings(request, null)).getStringList());
    this.fetchingStrings.next(false);
  }
}
