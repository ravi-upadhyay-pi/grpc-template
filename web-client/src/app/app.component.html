<mat-card class="create-string-container">
  <mat-card-title>Create new String</mat-card-title>
  <mat-card-content>
    <mat-form-field appearance="fill">
      <mat-label>String Id</mat-label>
      <input matInput type="text" [formControl]="newStringId">
    </mat-form-field>
    <mat-form-field appearance="fill">
      <mat-label>String Value</mat-label>
      <input matInput type="text" [formControl]="newStringValue">
    </mat-form-field>
  </mat-card-content>
  <mat-card-actions>
    <button mat-button (click)="saveString">Save String</button>
  </mat-card-actions>
</mat-card>

<mat-card class="existing-strings">
  <mat-card-title>Strings</mat-card-title>
  <mat-card-content>
    <mat-progress-bar 
    *ngIf="(fetchingStrings | async) else table" 
    mode="indeterminate">
  </mat-progress-bar>
  <ng-template #table>
    <table mat-table [dataSource]="(strings | async) ?? []">
      <ng-container matColumnDef="stringId">
        <th mat-header-cell *matHeaderCellDef>Id</th>
        <td mat-cell *matCellDef="let element"> {{element.getId()}} </td>
      </ng-container>
      <ng-container matColumnDef="stringValue">
        <th mat-header-cell *matHeaderCellDef>Value</th>
        <td mat-cell *matCellDef="let element"> {{element.getString()}} </td>
      </ng-container>
      <tr mat-header-row *matHeaderRowDef="['stringId', 'stringValue']"></tr>
      <tr mat-row *matRowDef="let row; columns: ['stringId', 'stringValue'];"></tr>
    </table>
    <mat-paginator 
      [length]="(existingStringsCount | async) ?? 0"
      [pageSize]="5"
      [pageSizeOptions]="[2, 5, 10]"
      (page)="updatePage($event)">
    </mat-paginator>
  </ng-template>
  </mat-card-content>
</mat-card>
