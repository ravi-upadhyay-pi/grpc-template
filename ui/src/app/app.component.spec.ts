import { TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { AppComponent } from './app.component';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { MatInputModule } from '@angular/material/input';
import { MatCardModule } from '@angular/material/card';
import { MatButtonModule } from '@angular/material/button';
import { MatTableModule } from '@angular/material/table';
import { MatPaginatorModule } from '@angular/material/paginator';
import { MatProgressBarModule } from '@angular/material/progress-bar';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [
        RouterTestingModule,
        BrowserAnimationsModule,
        ReactiveFormsModule,
        MatInputModule,
        MatCardModule,
        MatButtonModule,
        MatTableModule,
        MatPaginatorModule,
        MatProgressBarModule,
      ],
      declarations: [
        AppComponent
      ],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    expect(fixture.componentInstance).toMatchSnapshot();
  });
});
