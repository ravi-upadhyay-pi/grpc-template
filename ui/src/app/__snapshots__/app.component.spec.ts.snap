// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
AppComponent {
  "__ngContext__": 0,
  "client": GrpcTemplateClient {
    "client_": Z {
      "a": "text",
      "b": Array [],
      "c": null,
      "f": false,
      "g": false,
      "h": Array [],
    },
    "credentials_": Object {},
    "hostname_": "",
    "methodDescriptorGetStrings": E {
      "a": "unary",
      "b": [Function],
      "c": [Function],
      "f": [Function],
      "g": [Function],
      "name": "/grpc_template.GrpcTemplate/GetStrings",
    },
    "methodDescriptorGetStringsCount": E {
      "a": "unary",
      "b": [Function],
      "c": [Function],
      "f": [Function],
      "g": [Function],
      "name": "/grpc_template.GrpcTemplate/GetStringsCount",
    },
    "methodDescriptorSaveString": E {
      "a": "unary",
      "b": [Function],
      "c": [Function],
      "f": [Function],
      "g": [Function],
      "name": "/grpc_template.GrpcTemplate/SaveString",
    },
    "options_": Object {
      "format": "text",
    },
  },
  "existingStringsCount": BehaviorSubject {
    "_value": 0,
    "closed": false,
    "currentObservers": null,
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
  "fetchingStrings": BehaviorSubject {
    "_value": true,
    "closed": false,
    "currentObservers": null,
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
  "newStringId": FormControl2 {
    "_composedAsyncValidatorFn": null,
    "_composedValidatorFn": null,
    "_hasOwnPendingAsyncValidator": false,
    "_onChange": Array [],
    "_onCollectionChange": [Function],
    "_onDisabledChange": Array [],
    "_parent": null,
    "_pendingChange": false,
    "_pendingDirty": false,
    "_pendingTouched": false,
    "_pendingValue": "",
    "_rawAsyncValidators": null,
    "_rawValidators": null,
    "defaultValue": null,
    "errors": null,
    "pristine": true,
    "status": "VALID",
    "statusChanges": EventEmitter_ {
      "__isAsync": false,
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "touched": false,
    "value": "",
    "valueChanges": EventEmitter_ {
      "__isAsync": false,
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
  },
  "newStringValue": FormControl2 {
    "_composedAsyncValidatorFn": null,
    "_composedValidatorFn": null,
    "_hasOwnPendingAsyncValidator": false,
    "_onChange": Array [],
    "_onCollectionChange": [Function],
    "_onDisabledChange": Array [],
    "_parent": null,
    "_pendingChange": false,
    "_pendingDirty": false,
    "_pendingTouched": false,
    "_pendingValue": "",
    "_rawAsyncValidators": null,
    "_rawValidators": null,
    "defaultValue": null,
    "errors": null,
    "pristine": true,
    "status": "VALID",
    "statusChanges": EventEmitter_ {
      "__isAsync": false,
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
    "touched": false,
    "value": "",
    "valueChanges": EventEmitter_ {
      "__isAsync": false,
      "closed": false,
      "currentObservers": null,
      "hasError": false,
      "isStopped": false,
      "observers": Array [],
      "thrownError": null,
    },
  },
  "strings": BehaviorSubject {
    "_value": Array [],
    "closed": false,
    "currentObservers": null,
    "hasError": false,
    "isStopped": false,
    "observers": Array [],
    "thrownError": null,
  },
}
`;
